### PR TITLE
Use weights_only=True for remaining torch.load() calls

### PR DESCRIPTION
**Pull request by adrianlizarraga**
### Description

Follow-up to PR #28097. Applies the same `_torch_load_weights_only()` wrapper to the two remaining `torch.load()` call sites.

`torch.load` can deserialize arbitrary Python pickle payloads. Using `weights_only=True` restricts loading to tensor/checkpoint data on supported PyTorch versions and is the safer default. The wrapper gracefully falls back to the default `torch.load` behavior on older PyTorch versions that do not support the `weights_only` parameter.

### Summary of Changes

| File | Change |
|------|--------|
| `onnxruntime/test/testdata/test_data_generation/lr_scheduler/lr_scheduler_test_data_generator.py` | Adds `_torch_load_weights_only()` helper and uses it when loading scheduler/optimizer state dicts. |
| `orttraining/orttraining/test/python/orttraining_test_ortmodule_pytorch_ddp.py` | Adds `_torch_load_weights_only()` helper and uses it when loading DDP model checkpoint. |

### Motivation and Context

These were the last two `torch.load()` calls in the repository without `weights_only=True`. While both are in test/tooling code with low direct risk, this change ensures consistency with the pattern established in PR #28097 and eliminates all unsafe deserialization call sites.

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_pytorch_ddp.py [hunk 1]
```diff
@@ -15,6 +16,20 @@
...
+logger = logging.getLogger(__name__)
+
+
+def _torch_load_weights_only(path: str, **kwargs):
+    try:
+        return torch.load(path, weights_only=True, **kwargs)
+    except TypeError:
+        logger.warning(
+            "Current PyTorch version does not support torch.load(..., weights_only=True); "
+            "falling back to default torch.load behavior for %s.",
```
[the remark this answers is not in the document]
Comment (adrianlizarraga) [reply orttraining/orttraining/test/python/orttraining_test_ortmodule_pytorch_ddp.py]: Updated to use `inspect.signature`

File: onnxruntime/test/testdata/test_data_generation/lr_scheduler/lr_scheduler_test_data_generator.py [hunk 2]
```diff
@@ -4,9 +4,25 @@
...
+logger = logging.getLogger(__name__)
+
+
+def _torch_load_weights_only(path: str, **kwargs):
+    try:
+        return torch.load(path, weights_only=True, **kwargs)
+    except TypeError:
+        logger.warning(
+            "Current PyTorch version does not support torch.load(..., weights_only=True); "
+            "falling back to default torch.load behavior for %s.",
```
[the remark this answers is not in the document]
Comment (adrianlizarraga) [reply onnxruntime/test/testdata/test_data_generation/lr_scheduler/lr_scheduler_test_data_generator.py]: Updated to use `inspect.signature`

**Review by tianleiwu — APPROVED** [review]

**Review by tianleiwu — APPROVED** [review]
**APPROVE** — Clean, correct security hardening of the last two `torch.load()` call sites.

The `inspect.signature(torch.load)` probe at module level is a nice improvement over the `try/except TypeError` fallback used in PR #28097 — it avoids masking unrelated `TypeError` exceptions from `torch.load` itself (e.g., corrupted files or bad arguments). Both files correctly forward `**kwargs` and load payloads (state dicts) that are safe for the restricted `weights_only=True` loader.

**Nitpick (not blocking):** The repo now has two different implementations of the same helper — the PR #28097 files (`t5_helper.py`, `nv_run_pretraining.py`) still use `try/except TypeError`, while these new files use `inspect.signature`. Consider back-porting the improved probe to the older helpers in a follow-up for consistency, or extracting a shared utility.
